src/plugins/intel_gpu/src/plugin/remote_context.cpp

```diff
@@ -232,4 +237,25 @@ void RemoteContextImpl::check_if_shared() const {
     OPENVINO_ASSERT(m_type == ContextType::VA_SHARED, "[GPU] Shared context is required to to share this type of memory");
 }
 
+void RemoteContextImpl::initialize() {
+    if (m_is_initialized)
+        return;
```

Comment:
this is not thread-reentrant as multiple thread can go into this function concurrently. is it safe to do so? I think we need a lock here.

Reply:
Yeah, it makes sense, updated, thanks!